### PR TITLE
Misc updates to the Tables api

### PR DESCRIPTION
**Pull request by alissapajer**

### CODE REVIEW

**Comment by alissapajer** [comment]
As I've worked on the mock/impl, I think there actually could end up being a line we can draw between the `Tables` api and a new `Preparations` api. But I haven't fully worked that out yet so for now I've kept everything in `Tables`.

File: api/src/main/scala/quasar/api/table/Tables.scala [hunk 1]
```diff
@@ -17,25 +17,31 @@
...
+  def createTable(table: Table[Q]): F[CreationError \/ I]
 
-  def setTableAttributes(tableId: I, attributes: NonEmptyList[TableAttribute]): F[Condition[ModificationError[I]]]
+  def replaceTable(tableId: I, table: Table[Q]): F[Condition[ModificationError[I]]]
 
   def prepareTable(tableId: I): F[Condition[PrePreparationError[I]]]
 
   def preparationStatus(tableId: I): F[ExistenceError[I] \/ PreparationStatus]
 
+  def cancelPreparation(tableId: I): F[Condition[ExistenceError[I]]]
```
Comment (wemrysi) [inline api/src/main/scala/quasar/api/table/Tables.scala]: Noice.

File: api/src/main/scala/quasar/api/table/Tables.scala [hunk 2]
```diff
@@ -17,25 +17,31 @@
...
-import scalaz.{\/, NonEmptyList}
+import slamdata.Predef.Unit
+import scalaz.\/
 
 /** @tparam I identity
+  * @tparam Q query type
   * @tparam D materialized table data
   */
-trait Tables[F[_], G[_], I, D] {
+trait Tables[F[_], G[_], I, Q, D] {
```
Comment (wemrysi) [inline api/src/main/scala/quasar/api/table/Tables.scala]: Achievement Unlocked: Parametric AF.

File: api/src/main/scala/quasar/api/table/Table.scala [hunk 3]
```diff
@@ -16,4 +16,8 @@
 
 package quasar.api.table
 
-final case class Table(name: TableAttribute.Name, sql2: TableAttribute.Sql2)
+import slamdata.Predef.String
+
+final case class Table[Q](name: TableName, query: Q)
```
Comment (wemrysi) [inline api/src/main/scala/quasar/api/table/Table.scala]: :+1:

**Pull request merged**